domain/secret/service/params.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2024 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package service
+
+import (
+	"time"
+
+	"github.com/juju/juju/core/leadership"
+	"github.com/juju/juju/core/permission"
+	"github.com/juju/juju/core/secrets"
+)
+
+// CreateSecretParams are used to create a secret.
+type CreateSecretParams struct {
+	UpdateSecretParams
+	Version int
+
+	// One owner must be non-nil.
+
+	CharmOwner *CharmSecretOwner
+	ModelOwner *string
+}
+
+// UpdateSecretParams are used to update a secret.
+type UpdateSecretParams struct {
+	LeaderToken  leadership.Token
+	RotatePolicy *secrets.RotatePolicy
+	ExpireTime   *time.Time
+	Description  *string
+	Label        *string
+	Params       map[string]interface{}
+	Data         secrets.SecretData
+	ValueRef     *secrets.ValueRef
```

Comment:
And here cannot have both Data and ValueRef.

Reply:
Right. Validation and business logic not added in this PR though.